package-utils.el

```diff
@@ -116,21 +116,20 @@ See the second argument to `package-menu--generate'."
         (insert "(setq load-path (append load-path my-paths))\n")
         (insert "(dolist (p my-paths) "
                 "(byte-recompile-directory p 0 t))"))
-      (let ((proc
-             (make-process
-              :name "emacs-recompile-all-proc"
-              :filter print-fn
-              ;; Don't encode `stdout' as string.
-              :coding 'no-conversion
-              :command
-              (list emacs-binary
-                    "-Q" "--batch" "--script" temp-file "--kill")
-              :connection-type 'pipe)))
-        (while (accept-process-output proc)))
-      (delete-file temp-file))
-
-    (message "Recompiled: %d directories"
-             (length load-paths-to-compile))))
+      (make-process
+       :name "emacs-recompile-all-proc"
+       :filter print-fn
+       ;; Don't encode `stdout' as string.
+       :coding 'no-conversion
+       :command
+       (list emacs-binary
+             "-Q" "--batch" "--script" temp-file "--kill")
+       :connection-type 'pipe
+       :sentinel (lambda (process _msg)
+		   (when (eq (process-status process) 'exit)
+		     (message "Recompiled: %d directories"
+			      (length load-paths-to-compile))
+		     (delete-file temp-file)))))))
```

Comment:
Shouldn't `(delete-file temp-file)` be moved outside of the sentinel?

We want to delete the file no matter what happens to the process (instead of exit, it could fail or give some other kind of error).

Or, at least move the file removal out of the `exit` check.